module/VuFind/src/VuFind/Search/Blender/Params.php

```diff
@@ -0,0 +1,655 @@
+                    $params->addFilter("$translatedField:$defaultValue");
+                }
+            }
+        }
+    }
+
+    /**
+     * Proxy a method call to parent class and all backend params classes
+     *
+     * @param string $method Method
+     * @param array  $params Method parameters
+     *
+     * @return mixed
+     */
+    protected function proxyMethod(string $method, array $params)
+    {
+        $result = call_user_func_array(['parent', $method], $params);
+        foreach ($this->searchParams as $searchParams) {
+            $result = call_user_func_array([$searchParams, $method], $params);
+        }
+        return $result;
+    }
+
+    /**
+     * Translate a facet
+     *
+     * @param string $field     Facet field
+     * @param string $backendId Backend ID
+     *
+     * @return string
+     */
+    protected function translateFacet(string $field, string $backendId): string
+    {
+        $prefix = '';
+        if (substr($field, 0, 1) === '~') {
+            $field = substr($field, 1);
+            $prefix = '~';
+        }
+        $fieldConfig = $this->mappings['Facets']['Fields'][$field] ?? [];
+        $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+        return !empty($mappings['Field']) ? ($prefix . $mappings['Field']) : '';
+    }
+
+    /**
+     * Check if the filter is a special Blender filter
+     *
+     * @param string $filter Filter
+     *
+     * @return bool
+     */
+    protected function isBlenderFilter(string $filter): bool
+    {
+        [$field] = $this->parseFilter($filter);
+        if (substr($field, 0, 1) === '~') {
+            $field = substr($field, 1);
+        }
+        return 'blender_backend' === $field;
+    }
+
+    /**
+     * Translate a filter
+     *
+     * @param string $filter    Filter
+     * @param string $backendId Backend ID
+     *
+     * @return array
+     */
+    protected function translateFilter(string $filter, string $backendId): array
+    {
+        [$field, $value] = $this->parseFilter($filter);
+        $prefix = '';
+        $firstChar = substr($field, 0, 1);
+        if (in_array($firstChar, ['~', '-'])) {
+            $prefix = $firstChar;
+            $field = substr($field, 1);
+        }
+
+        $fieldConfig = $this->mappings['Facets']['Fields'][$field] ?? [];
+        if ($ignore = $fieldConfig['Mappings'][$backendId]['Ignore'] ?? '') {
+            if (true === $ignore || in_array($value, (array)$ignore)) {
+                return [null];
+            }
+        }
+        $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+        $translatedField = $mappings['Field'] ?? '';
+        if (!$mappings || !$translatedField) {
+            // Facet not supported by the backend
+            return [];
+        }
+
+        // Map filter value
+        $facetType = $fieldConfig['Type'] ?? 'normal';
+        if ('boolean' === $facetType) {
+            $value = (bool)$value;
+        }
+        $resultValues = [];
```

Comment:
This is a long method... might the creation of `$resultValues` be a piece that could be refactored to a support method for readability?

Reply:
Yes, done, but I made the split a bit differently.